server/Controllers/ControlPCController.cs

```diff
@@ -0,0 +1,142 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using WAMServer.Interfaces;
+using WAMServer.Models;
+using WAMServer.Records.Bodies;
+
+namespace WAMServer.Controllers
+{
+    /// <summary>
+    /// The ControlPC controller. Responsible for managing ControlPC records.
+    /// </summary>
+    [Route("api/[controller]")]
+    [ApiController]
+    public class ControlPCController : ControllerBase
+    {
+        private readonly IRepository<ControlPC> _controlPCRepository;
+        private readonly IRepository<User> _userRepository;
+
+        private readonly IControlPC<ControlPC> _controlPCRepository2;
+
+        /// <summary>
+        /// The constructor of the ControlPC controller.
+        /// </summary>
+        /// <param name="controlPCRepository">The ControlPC repository.</param>
+        /// <param name="userRepository">The user repository.</param>
+        public ControlPCController(IRepository<ControlPC> controlPCRepository, IRepository<User> userRepository)
+        {
+            _controlPCRepository = controlPCRepository;
+            _userRepository = userRepository;
+        }
+
+        /// <summary>
+        /// Gets all ControlPC records.
+        /// </summary>
+        /// <returns>The list of ControlPC records.</returns>
+        [HttpGet]
+        [Authorize]
+        public async Task<ActionResult<IEnumerable<ControlPC>>> GetControlPCs()
+        {
+            var controlPCs = await _controlPCRepository.GetAllAsync();
+            return Ok(controlPCs);
+        }
+
+        /// <summary>
+        /// Gets a ControlPC record by its ID.
+        /// </summary>
+        /// <param name="id">The ID of the ControlPC record.</param>
+        /// <returns>The ControlPC record.</returns>
+        [HttpGet("{id}")]
+        [Authorize]
+        public async Task<ActionResult<ControlPC>> GetControlPC(Guid id)
+        {
+            var controlPC = await _controlPCRepository2.GetAsync(id);
+            if (controlPC == null)
+            {
+                return NotFound();
+            }
+            return Ok(controlPC);
+        }
+
+        /// <summary>
+        /// Creates a new ControlPC record.
+        /// </summary>
+        /// <param name="controlPC">The ControlPC object.</param>
+        /// <returns>The created ControlPC record.</returns>
+        [HttpPost]
+        [Authorize]
+        public async Task<ActionResult<ControlPC>> CreateControlPC([FromBody] ControlPC controlPC)
+        {
+            if (controlPC == null)
+            {
+                return BadRequest(new { message = "Invalid data." });
```

Comment:
Hiervoor is een record. Check WAMServer.Records.Bodies.ErrorBody

Reply:
heb het verbeterd met de ErrorBody